addons/website/static/src/components/wysiwyg_adapter/wysiwyg_adapter.js

```diff
@@ -159,12 +159,24 @@ export class WysiwygAdapterComponent extends Wysiwyg {
                         hasFakeState = true;
                     },
                     onLeave: () => history.back(),
-                    reloadIframe: false
+                    reloadIframe: true
                 });
             };
-            window.addEventListener('popstate', leaveOnBackNavigation);
+
+            const handleBeforePopstate = (ev) => {
+                ev = ev.detail;
+                const action = ev.state?.nextState?.action;
+                if (action === "website" && document.querySelector(".editor_enable")) {
+                    leaveOnBackNavigation();
+                    router.skipLoad = true;
+                }
+            };
+
+            routerBus.removeEventListener("before_popstate", handleBeforePopstate);
+            routerBus.addEventListener("before_popstate", handleBeforePopstate);
+
             return () => {
-                window.removeEventListener('popstate', leaveOnBackNavigation);
+                routerBus.removeEventListener("before_popstate", handleBeforePopstate);
```

Comment:
The cleanup function in useEffect does removed the eventListner, then why are we removing it again and then adding it ? Is there any specific reason ?